src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsBuildPlanCreatorProvider.java

```diff
@@ -0,0 +1,33 @@
+package de.tum.in.www1.artemis.service.connectors.jenkins;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import de.tum.in.www1.artemis.domain.enumeration.ProgrammingLanguage;
+
+@Profile("jenkins")
+@Component
+public class JenkinsBuildPlanCreatorProvider {
+
+    private final JavaJenkinsBuildPlanCreator javaJenkinsBuildPlanCreator;
+
+    public JenkinsBuildPlanCreatorProvider(JavaJenkinsBuildPlanCreator javaJenkinsBuildPlanCreator) {
+        this.javaJenkinsBuildPlanCreator = javaJenkinsBuildPlanCreator;
+    }
+
+    /**
+     * Gives a Jenkins plan builder, that is able to build plan configurations for the specified programming language
+     *
+     * @param programmingLanguage The programming language for which a build plan should get created
+     * @return The configuration builder for the specified language
+     * @see JavaJenkinsBuildPlanCreator
+     */
+    public JenkinsXmlConfigBuilder builderFor(ProgrammingLanguage programmingLanguage) {
+        switch (programmingLanguage) {
+        case JAVA:
```

Comment:
A nit-pick here as well. Can you indent case and default statements more to the right side please?

Reply:
Sadly, I can't change this. Just like with our streams, our ~horrible~ great auto formatting reverts this on every commit.